packages/Core/iterable.candy

```diff
@@ -31,10 +32,10 @@ isEmpty iterable :=
 generateWithStateHelper = { generateWithStateHelper state mapper ->
   {
     mapper state | result.map { result ->
-      [newItem, newState] = result
+      [item, state] = result
```

Comment:
Just noticing that we could also inline the entire function using the `recursive` function.

Reply:
We could also use that for the remaining helper functions in this file, but I'd wait for better pattern support (#295) before doing so